sdks/java/container/build.gradle

```diff
@@ -94,8 +86,12 @@ licenseReport {
 task copyThirdPartyLicenses(type: Copy) {
   from ("third_party_licenses")
   into "build/target/third_party_licenses"
+  doLast {
+    file('third_party_licenses').deleteDir()
+  }
 }
 
+def Boolean pull_licenses
```

Comment:
Why pull this out into a global variable?

Reply:
when we pass `docker-pull-licenses` to `:sdks:java:container:docker`, `generateThirdPartyLicenses` task doesn't understand it. In order to pass the tag from docker task to `generateThirdPartyLicenses`, we need a global value to share it.

Reply:
I tried changing `generateThirdPartyLicenses` to read `project.rootProject.hasProperty(["docker-pull-licenses"])` directly, and it seemed to work fine when I called `./gradlew :sdks:java:container:docker -Pdocker-pull-licenses`.

Reply:
(Sorry for picking nits by the way, I'm just trying to understand Gradle better 😅)

Reply:
That was the behavior I observed when I implement it. But yesterday, when I try it again, it didn't work as expected. So I introduced the pull_licenses variable. But today, as you pointed out, I tried again, it worked! I reversed to original code.